libethash-cuda/CUDAMiner.cpp

```diff
@@ -509,14 +509,10 @@ void CUDAMiner::search(
 				for (unsigned int j = 0; j < found_count; j++) {
 					nonces[j] = nonce_base + buffer->result[j].gid;
 					if (s_noeval) {
-						mixes[j][0] = buffer->result[j].mix[0];
-						mixes[j][1] = buffer->result[j].mix[1];
-						mixes[j][2] = buffer->result[j].mix[2];
-						mixes[j][3] = buffer->result[j].mix[3];
-						mixes[j][4] = buffer->result[j].mix[4];
-						mixes[j][5] = buffer->result[j].mix[5];
-						mixes[j][6] = buffer->result[j].mix[6];
-						mixes[j][7] = buffer->result[j].mix[7];
+						uint32_t* m = (uint32_t *)(mixes + j);
```

Comment:
Honesty, WTF is going on here?

Reply:
Honestly is right!!!

What is needed here is a word wise copy to achieve the converson from uint32 array to h256. The previous code left that for later! But using the expression further down stopped working when Ubuntu updated to GCC 5.4.0!!! Not sure why?

Reply:
Probably it recognized the undefined behavior. Just use memcpy to copy it. It will handle memory alignment issues.

Reply:
No point, we know this is is word aligned, and a word mutiple. This is faster in a critical loop.